packages/pretty-format/src/index.js

```diff
@@ -57,15 +59,15 @@ function printFunction(val, printFunctionName) {
   }
 }
 
-function printSymbol(val) {
+function printSymbol(val: Symbol): string {
   return symbolToString.call(val).replace(SYMBOL_REGEXP, 'Symbol($1)');
 }
 
-function printError(val) {
+function printError(val: Error): string {
   return '[' + errorToString.call(val) + ']';
 }
 
-function printBasicValue(val, printFunctionName, escapeRegex) {
+function printBasicValue(val: any, printFunctionName: boolean, escapeRegex: boolean): StringOrNull {
```

Comment:
Yup, this is what I would've expected. When I was writing pretty-format I did all sorts of micro-optimizations like storing `typeof val` as a variable. Flow also requires things to be stringified in a certain way (i.e. `'' + bool` is not considered valid).

We don't really want to modify the code of pretty-format by much because the optimizations do matter to make it as fast as `JSON.stringify`.

The only remaining option would be to type cast values in different places:

```js
const typeOf = typeof val;

if (typeOf === 'number') { 
  return printNumber(((val: any): number));
}
```

```js
if (val === true || val === false) {
  return '' + ((val: any): string);
}
```

These will then get stripped away, leaving the original code.